core/ops.rs

```diff
@@ -289,3 +290,69 @@ impl DerefMut for OpState {
     &mut self.gotham_state
   }
 }
+
+use crate::op2;
+use crate::Resource;
+use std::borrow::Cow;
+use std::mem::transmute;
+
+pub struct WasmMemory(Option<*const v8::fast_api::FastApiTypedArray<u8>>);
+
+struct WasmMemoryResource(NonNull<v8::WasmMemoryObject>);
+
+impl Resource for WasmMemoryResource {
+  fn name(&self) -> Cow<'static, str> {
+    "wasmMemory".into()
+  }
+}
+
+#[op2(core)]
+#[smi]
+pub fn op_core_set_wasm_memory(
+  state: &mut OpState,
+  #[global] memory: v8::Global<v8::WasmMemoryObject>,
+) -> ResourceId {
+  state
+    .resource_table
+    .add(WasmMemoryResource(memory.into_raw()))
+}
+
+impl WasmMemory {
+  #[inline]
+  pub fn new(mem: *const v8::fast_api::FastApiTypedArray<u8>) -> Self {
+    Self(Some(mem))
+  }
+
+  #[inline]
+  pub fn null() -> Self {
+    Self(None)
+  }
+
+  #[inline]
+  pub fn get<'s>(
+    self,
+    state: &mut OpState,
+    rid: ResourceId,
+  ) -> Option<&'s mut [u8]> {
+    match self.0 {
+      Some(mem) => unsafe { (*mem).get_storage_if_aligned() },
+      None => {
+        let resource =
+          state.resource_table.get::<WasmMemoryResource>(rid).ok()?;
+        // SAFETY: `v8::Local` is always non-null pointer; the `HandleScope` is
+        // already on the stack, but we don't have access to it.
```

Comment:
This code worries me a bit because there's no guarantee that the scope is on the stack when it is called -- it is possible to create one of these via WasmMemory::null() and then call get on it.

As mentioned above, it would be better to hide this inside the op itself because we can make those guarantees there.